experiments/conditions.go

```diff
@@ -0,0 +1,54 @@
+package experiments
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	"github.com/argoproj/argo-rollouts/utils/conditions"
+)
+
+func (ec *ExperimentController) calculateExperimentConditions(experiment *v1alpha1.Experiment, newStatus v1alpha1.ExperimentStatus, templateRSs map[string]*appsv1.ReplicaSet) v1alpha1.ExperimentStatus {
+	switch {
+	case conditions.ExperimentCompleted(newStatus):
```

Comment:
Just a suggestion.

Looks like `conditions.ExperimentCompleted`, `conditions. ExperimentProgressing` should never return true simultaneously for the same experiment. Only in this case it is safe to use these functions in switch expression. If, because of a bug, more than one return true than only first `case` got executed.

I would consider creating `ExperimentStatusCode` type which can take values `ExperimentCompleted`, `ExperimentProgressing` etc and single method `GetExperimentStatus` which can be used in `calculateExperimentConditions`.  You can even consider saving `statusCode`  in CRD status to simplify health assessment.